openfl/component/aggregator/aggregator.py

```diff
@@ -632,7 +669,22 @@ def send_local_task_results(
             task_results.append(tensor_key)
 
         self.collaborator_tasks_results[task_key] = task_results
-        self._end_of_task_check(task_name)
+
+        self._is_collaborator_done(collaborator_name)
+
+        # If minimum required collaborators have reported results,
+        # then identify stragglers and call for early round end.
+        if self.straggler_handling_policy.straggler_cutoff_check(
+            len(self.collaborators_done), len(self.authorized_cols)
+        ):
+            self.stragglers = [
+                collab_name
+                for collab_name in self.authorized_cols
+                if collab_name not in self.collaborators_done
+            ]
+            if len(self.stragglers) != 0:
+                self.logger.warning(f"Identified straggler collaborators: {self.stragglers}")
+            self._end_of_round_check()
```

Comment:
This code is an almost exact duplicate of lines 419-434. Would it make sense to extract it in a `self._end_of_round_with_stragglers_check()` method?

Reply:
Consolidated duplicate code by updating it into the `_end_of_round_with_stragglers_check()` method.